src/components/structures/auth/Login.tsx

```diff
@@ -117,7 +117,7 @@ export default class LoginComponent extends React.PureComponent<IProps, IState>
 
             flows: null,
 
-            username: "",
+            username: props.defaultUsername? props.defaultUsername: '',
```

Comment:
Please add `defaultUsername` as an optional prop to `IProps` above.

Reply:
done